File: VB/CustomItemExtension/CustomItems/TreeList/TreeListItemExtensionModule.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Ribbon

Namespace DevExpresss.DashboardWin.CustomItemExtension
	Public Class TreeListItemExtensionModule
		Private dashboardControl As IDashboardControl
		Public Sub AttachViewer(ByVal viewer As DashboardViewer)
			AttachDashboardControl(viewer)
		End Sub
		Public Sub DetachViewer()
			Detach()
		End Sub
		Public Sub AttachDesigner(ByVal designer As DashboardDesigner)
			AttachDashboardControl(designer)
			designer.CreateCustomItemBars(GetType(TreeListItemMetadata))
			RemoveDrillDownBarItem(designer)
		End Sub
		Public Sub DetachDesigner()
			Detach()
		End Sub
		Private Sub Detach()
			If dashboardControl IsNot Nothing Then
				RemoveHandler dashboardControl.CustomDashboardItemControlCreating, AddressOf OnCustomDashboardItemControlCreating
			End If
		End Sub
		Private Sub AttachDashboardControl(ByVal dashboardControl As IDashboardControl)
			If dashboardControl IsNot Nothing Then
				Me.dashboardControl = dashboardControl
				dashboardControl.CalculateHiddenTotals = True
				AddHandler dashboardControl.CustomDashboardItemControlCreating, AddressOf OnCustomDashboardItemControlCreating
			End If
		End Sub
		Private Sub OnCustomDashboardItemControlCreating(ByVal sender As Object, ByVal e As CustomDashboardItemControlCreatingEventArgs)
			Dim dashboardControl As IDashboardControl = DirectCast(sender, IDashboardControl)
			If e.MetadataType Is GetType(TreeListItemMetadata) Then
				e.CustomControlProvider = New TreeListItemControlProvider(TryCast(dashboardControl.Dashboard.Items(e.DashboardItemName), CustomDashboardItem(Of TreeListItemMetadata)))
			End If
		End Sub
		Private Sub RemoveDrillDownBarItem(ByVal designer As DashboardDesigner)
			Dim page As RibbonPage = designer.Ribbon.GetDashboardRibbonPage(GetType(TreeListItemMetadata), DashboardRibbonPage.Data)
			Dim interactivityGroup As RibbonPageGroup = page.Groups(1)
			Dim drillDownBarItem As BarItem = interactivityGroup.ItemLinks(2).Item
			interactivityGroup.ItemLinks.Remove(drillDownBarItem)
		End Sub

	End Class
End Namespace

File: CS/CustomItemExtension/CustomItems/TreeList/TreeListItemExtensionModule.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Ribbon;

namespace DevExpresss.DashboardWin.CustomItemExtension
{
    public class TreeListItemExtensionModule {
        IDashboardControl dashboardControl;
        public void AttachViewer(DashboardViewer viewer) {
            AttachDashboardControl(viewer);
        }
        public void DetachViewer() {
            Detach();
        }
        public void AttachDesigner(DashboardDesigner designer) {
            AttachDashboardControl(designer);
            designer.CreateCustomItemBars(typeof(TreeListItemMetadata));
            RemoveDrillDownBarItem(designer);
        }
        public void DetachDesigner() {
            Detach();
        }
        void Detach() {
            if(dashboardControl != null)
                dashboardControl.CustomDashboardItemControlCreating -= OnCustomDashboardItemControlCreating;
        }
        void AttachDashboardControl(IDashboardControl dashboardControl) {
            if(dashboardControl != null) {
                this.dashboardControl = dashboardControl;
                dashboardControl.CalculateHiddenTotals = true;
                dashboardControl.CustomDashboardItemControlCreating += OnCustomDashboardItemControlCreating;
            }
        }
        void OnCustomDashboardItemControlCreating(object sender, CustomDashboardItemControlCreatingEventArgs e) {
            IDashboardControl dashboardControl = (IDashboardControl)sender;
            if(e.MetadataType == typeof(TreeListItemMetadata))
                e.CustomControlProvider = new TreeListItemControlProvider(dashboardControl.Dashboard.Items[e.DashboardItemName] as CustomDashboardItem<TreeListItemMetadata>);
        }
        void RemoveDrillDownBarItem(DashboardDesigner designer) {
            RibbonPage page = designer.Ribbon.GetDashboardRibbonPage(typeof(TreeListItemMetadata), DashboardRibbonPage.Data);
            RibbonPageGroup interactivityGroup = page.Groups[1];
            BarItem drillDownBarItem = interactivityGroup.ItemLinks[2].Item;
            interactivityGroup.ItemLinks.Remove(drillDownBarItem);
        }

    }
}
